### DISCUSSION TITLE
Modular Monolith - A Gentle Introduction | Dan Does Code

### DISCUSSION THREAD

**Discussion by giscus** [danielmackay/nextjs-personal-blog#135] [Blog Comments]
# Modular Monolith - A Gentle Introduction | Dan Does Code

When starting a new project, one of the first and most important decisions you'll have to make is choosing a Software Architecture. Choose an architecture that is too simple and the project will quickly outgrow it. Choose an architecture that is too complex, and you'll spend more time maintaining the architecture than building the project.  Therefore, it's important to choose an architecture that is 'just right'. Introducing Modular Monoliths — "The Goldilocks Architecture".

https://www.dandoescode.com/blog/modular-monolith/a-gentle-introduction

**Comment by Hona** [comment]
Insightful article, Dan!

Can't wait for more :D
